lib/sidekiq_publisher/client.rb

```diff
@@ -4,29 +4,77 @@
 
 module SidekiqPublisher
   class Client < Sidekiq::Client
+    # A client that can be used to bulk-publish jobs of many different types together in
+    # a single atomic push. Unfortunately, Sidekiq does not provide an API to do this --
+    # their bulk API only supports an array of args for a single job class.
+    #
+    # This subclass originally relied on a handful of internal private methods. Some of
+    # those internal methods are now replicated here, in ezCater-maintained code, to help
+    # prevent future minor or patch updates to sidekiq from breaking this class.
+
     def bulk_push(items)
       payloads = items.map do |item|
         normed = normalize_item(item)
-        process_single(item["class"], normed) || nil
-      end.compact
+        result = middleware.invoke(item["class"], normed, normed["queue"], @redis_pool) { normed }
+        result || nil
+      end
+
+      raw_multi_push(payloads.compact)
+    end
+
+    private
 
+    # A copy of sidekiq's `raw_push` with the following changes:
+    #   * returns the number of payloads pushed
+    #   * uses `multi` instead of `pipelined` so that the entire batch is atomic together
+    #     (see: https://github.com/ezcater/sidekiq_publisher/pull/5#discussion_r188069548)
+    def raw_multi_push(payloads)
```

Comment:
The current implementation of Sidekiq's `raw_push` can be found [here](https://github.com/mperham/sidekiq/blob/v6.4.2/lib/sidekiq/client.rb#L197).